game/src/pregame.rs

```diff
@@ -24,8 +27,12 @@ pub struct TitleScreen {
 
 impl TitleScreen {
     pub fn new(ctx: &mut EventCtx, app: &App) -> TitleScreen {
-        let mut rng = app.primary.current_flags.sim_flags.make_rng();
-        TitleScreen {
+       let mut rng = app.primary.current_flags.sim_flags.make_rng();
+       let scenario_generator = ScenarioGenerator::small_run(&app.primary.map);
+       let mut timer = Timer::new("title screen timer");
+       let _scenario = scenario_generator.generate(&app.primary.map, &mut rng, &mut timer);
```

Comment:
@dabreegster How do I say that the scenario should run? When I start the game, I don't see any traffic. Furthermore, I  see a  bunch of warnings and I'm not entirely sure why they happen. An excerpt is below:
```
119 warnings:
Can't spawn a Bike ending at border DirectedRoadID(468, forwards); no appropriate lanes there
Can't end_at_border for DirectedRoadID(468, forwards) without a sidewalk
Couldn't fulfill SpawnOverTime { num_agents: 10, start_time: Time(0.0), stop_time: Time(5.0), goal: EndOfRoad(DirectedRoadID { id: RoadID(468), forwards: true }), percent_driving: 0.5, percent_biking: 0.5, percent_use_transit: 0.5 } at all
Can't spawn a Bike ending at border DirectedRoadID(468, forwards); no appropriate lanes there
Can't end_at_border for DirectedRoadID(468, forwards) without a sidewalk
Couldn't fulfill SpawnOverTime { num_agents: 10, start_time: Time(0.0), stop_time: Time(5.0), goal: EndOfRoad(DirectedRoadID { id: RoadID(468), forwards: true }), percent_driving: 0.5, percent_biking: 0.5, percent_use_transit: 0.5 } at all
Can't end_at_border for DirectedRoadID(468, forwards) without a sidewalk
```
There are more warnings but there's a lot of them so I just put a few.

Reply:
You also need to do:

```
                scenario.instantiate(
                    &mut app.primary.sim,
                    &app.primary.map,
                    &mut rng,
                    timer,
                );
                app.primary
                    .sim
                    .tiny_step(&app.primary.map, &mut app.primary.sim_cb);
```

Reply:
Thanks for the suggestion! I got traffic going across now. I see a few options going forward:
- Have the title screen keep bouncing across a world filled with traffic
- Have the camera initially bounce randomly but settle on a random agent that wanders into the camera view
- Have the camera follow a random agent (remove bouncing behaviour entirely)
Which one do you like best?

Reply:
Awesome! No preference -- what do you think would make a good title screen? I can't imagine most people watch it for long. The second impl may be tricky (you'd have to do something with `AgentCache`, and if it complicates that code just for this case, probably not worth it). If you go with the third, you have to decide what to do after that agent is done.

(And I was wrong, you don't need `tiny_step`, unless you're going to immediately look for an agent to snap to. It just advances the sim 0.1s)

Reply:
Hmm, so when I tried removing `tiny_step`, the simulation either isn't running or it's running so slow that it's not visually interesting. With regard to those options, I think I'll abstain from the second option for now and I'll pursue the first and third options. That being said, how would I go about selecting a random agent and detecting when it's trip is done?

Reply:
I would try mimicking what `SpeedSetting::Fast` does in sandbox/speed. When you get an update event, run the sim forward a little. You can play with the speed -- 5x probably looks a little better.

For picking someone, you could do `sim.get_all_people`. I have a meeting in a moment, but check out info/person and info/trip to map person->trips and see trip status. (Or actually look at `sim.active_agents` maybe)